config/role.yaml

```diff
@@ -4,16 +4,103 @@ metadata:
   creationTimestamp: null
   name: knative-serving-operator
 rules:
+- apiGroups:
+  - '*'
+  resources:
+  - '*'
+  verbs:
+  - '*'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: knative-serving-operator-aggregated
+aggregationRule:
+  clusterRoleSelectors:
+# This (along with escalate below) allows the Operator to pick up any
+# roles that are provided to the admin of the cluster by knative serving
+# automatically.
+  - matchExpressions:
+      - {key: serving.knative.dev/release, operator: Exists}
+rules: [] # Rules are automatically filled in by the controller manager.
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: knative-serving-operator
+rules:
```

Comment:
Are these rules manually curated?

Reply:
Yes.

In general, only the `bind` section should need modification (if we add new non-Knative managed roles to the rolebindings used in Serving)

Reply:
Copied from below:

The only remaining diff between the role here and in eventing are the replacement of some labels,
a few explicit 'bind's (which now need to be added to roles the Operator binds to but
does not create), and the resource deletion section at the end.

In the future, if we move to a job based cleanup, deletion roles can be defined one-off.
If the Operator repos and/or controllers are to merge, this dramatically simplifies the collapse of
the roles needed.